cmd/gaia/cli_test/cli_test.go

```diff
@@ -139,6 +140,59 @@ func TestGaiaCLICreateValidator(t *testing.T) {
 	assert.Equal(t, "1/1", validator.PoolShares.Amount.String())
 }
 
+func TestGaiaCLISubmitProposal(t *testing.T) {
+
+	tests.ExecuteT(t, "gaiad unsafe_reset_all")
+	pass := "1234567890"
+	executeWrite(t, "gaiacli keys delete foo", pass)
+	executeWrite(t, "gaiacli keys delete bar", pass)
+	chainID := executeInit(t, "gaiad init -o --name=foo")
+	executeWrite(t, "gaiacli keys add bar", pass)
+
+	// get a free port, also setup some common flags
+	servAddr, port, err := server.FreeTCPAddr()
+	require.NoError(t, err)
+	flags := fmt.Sprintf("--node=%v --chain-id=%v", servAddr, chainID)
+
+	// start gaiad server
+	proc := tests.GoExecuteT(t, fmt.Sprintf("gaiad start --rpc.laddr=%v", servAddr))
+	defer proc.Stop(false)
+	tests.WaitForStart(port)
+
+	fooAddr, _ := executeGetAddrPK(t, "gaiacli keys show foo --output=json")
+	fooCech, err := sdk.Bech32ifyAcc(fooAddr)
+	require.NoError(t, err)
+
+	fooAcc := executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(50), fooAcc.GetCoins().AmountOf("steak"))
+
+	executeWrite(t, fmt.Sprintf("gaiacli gov submitproposal %v --proposer=%v --deposit=5steak --type=Text --title=Test --description=test --name=foo", flags, fooCech), pass)
+	time.Sleep(time.Second * 2) // waiting for some blocks to pass
+
+	fooAcc = executeGetAccount(t, fmt.Sprintf("gaiacli account %v %v", fooCech, flags))
+	assert.Equal(t, int64(45), fooAcc.GetCoins().AmountOf("steak"))
+
+	proposal1 := executeGetProposal(t, fmt.Sprintf("gaiacli gov queryproposal --proposalID=1 --output=json %v", flags))
+	assert.Equal(t, int64(1), proposal1.ProposalID)
+	assert.Equal(t, "DepositPeriod", proposal1.Status)
+
+	executeWrite(t, fmt.Sprintf("gaiacli gov deposit %v --depositer=%v --deposit=10steak --proposalID=1 --name=foo", flags, fooCech), pass)
+	time.Sleep(time.Second * 2) // waiting for some blocks to pass
```

Comment:
Can we use `WaitForHeight` instead? I think these sleeps may be the cause of CI nondeterminism, e.g. https://github.com/cosmos/cosmos-sdk/issues/1235

Reply:
This seems like a larger issue than this PR, as all of the CLI tests use time.sleep, not just the governance one, so this should probably be tackled separately.  Not sure how to use tests.WaitForHeight(), as I don't know the current height (there's no resultTx for me to get it from)

Reply:
This should be using `tests.WaitForHeight`. All of the CLI tests have been / should have been refactored to use this. Waiting a precise amount of seconds causes issues. `tests.WaitForHeight()` handles the pinging for a given height, to avoid the non-determinism. (https://github.com/cosmos/cosmos-sdk/blob/develop/tests/util.go)

If you want some "randomness" in the number of blocks to pass, timing isn't the way to go either. Use a random prng to specify the number of blocks, and then wait that precise number of blocks. Since you don't know the current block number, use `tests.WaitForNextHeightTM`. Depending on exact amounts of time causes issues, and this has been solved elsewhere.